tuf_conformance/repository_simulator.py

```diff
@@ -274,41 +292,30 @@ def fetch_metadata(self, role: str, version: int | None = None) -> bytes:
 
         If version is None, non-versioned metadata is being requested.
         """
+
         # decode role for the metadata
         role = parse.unquote(role, encoding="utf-8")
 
         if role == Root.type:
-            # return a version previously serialized in publish_root()
-            if version is None or version > len(self.signed_roots):
+            if version is None or version > len(self.signed_mds[Root.type]):
                 raise ValueError(f"Unknown root version {version}")
-            logger.debug("fetched root version %d", version)
-            return self.signed_roots[version - 1]
-
-        # sign and serialize the requested metadata
-        md = self.mds.get(role)
-
-        if md is None:
-            raise ValueError(f"Unknown role {role}")
-
-        md.signatures.clear()
-        for signer in self.signers[role].values():
-            md.sign(signer, append=True)
-
-        logger.debug(
-            "fetched %s v%d with %d sigs",
-            role,
-            md.signed.version,
-            len(self.signers[role]),
-        )
-        return md.to_bytes(JSONSerializer())
+            return self.signed_mds[Root.type][version - 1]
+        # Non-root mds:
+        if len(self.signed_mds[role]) == 0:
+            raise ValueError(f"The repository has not published metadata for '{role}'")
+        return self.signed_mds[role][-1]
```

Comment:
The repository gets a lot of calls for non-versioned, non-root roles. The only way I have made it work is to return the latest version for non-root roles. I haven't found a way to make it work when we return `self.signed_mds[role][version - 1]`. On the same note, since `Root` throws an error if the caller does not specify a version, and non-root roles don't, I have also not found a way to merge the two paths.

Reply:
There is two issues here: 
* the fixing the separate code paths is a suggestion, feel free to not do it
* We still should not answer with `self.signed_mds[role][-1]` _when the caller is asking for a specific version_. This is a bug


> since Root throws an error if the caller does not specify a version, and non-root roles don't

I don't think we need to raise in that root case, we can probably just make it work like the other roles: return the latest version

Reply:
I will get back to this. Keeping it open for now.